cpp/src/io/parquet/page_enc.cu

```diff
@@ -433,8 +433,9 @@ __global__ void __launch_bounds__(128)
                                   max_RLE_page_size(col_g.num_rep_level_bits(), num_vals));
 
       if (num_rows >= ck_g.num_rows ||
-          (values_in_page > 0 && (page_size + fragment_data_size > this_max_page_size)) ||
-          rows_in_page + frag_g.num_rows > max_page_size_rows) {
+          (values_in_page > 0 && (page_size + fragment_data_size > this_max_page_size ||
+                                  rows_in_page + frag_g.num_rows > max_page_size_rows)) ||
+          rows_in_page >= max_page_size_rows) {
```

Comment:
To be honest, I feel the entire of this `if` condition is too complex and error-prone. Can we break it into multiple conditions? For example:
```
auto const name1 = num_rows >= ck_g.num_rows;
auto const name2 = ....;
....

if(name1 && name2 || ....) {
...
}
```